crates/multi_buffer/src/multi_buffer.rs

```diff
@@ -3604,11 +3610,33 @@ impl MultiBufferSnapshot {
             } else {
                 range.end.row + 1
             };
+
+            let word_diffs = (!hunk.base_word_diffs.is_empty()
+                || !hunk.buffer_word_diffs.is_empty())
+            .then(|| {
+                let hunk_start_offset =
+                    Anchor::in_buffer(excerpt.id, excerpt.buffer_id, hunk.buffer_range.start)
+                        .to_offset(self);
+
+                hunk.base_word_diffs
+                    .iter()
+                    .map(|diff| {
+                        MultiBufferOffset(diff.start) + hunk_start_offset
+                            ..MultiBufferOffset(diff.end) + hunk_start_offset
```

Comment:
nit but this should work, reducing the need to wrap multi buffer offsets (which can end up in logic bugs)
```suggestion
                        hunk_start_offset + diff.start
                            ..hunk_start_offset + diff.end
```